From a3d22c96ec5eb60acf313f575c857427c11d6711 Mon Sep 17 00:00:00 2001
Message-ID: <a3d22c96ec5eb60acf313f575c857427c11d6711.1740173176.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1740173176.git.sharlatanus@gmail.com>
References: <cover.1740173176.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Mon, 3 Feb 2025 16:24:53 +0000
Subject: [PATCH 01/51] gnu: casacore: Update to 3.6.1.

* gnu/packages/astronomy.scm (casacore): Update to 3.6.1.
[argument] <phases>: Add 'hide-gfortran.
[inputs]: Add gsl.

Change-Id: I702427857cf001d583b775189f9d8f5d35fc7183
---
 gnu/packages/astronomy.scm | 18 +++++++++++++++---
 1 file changed, 15 insertions(+), 3 deletions(-)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index 52c295d98ad..1cd4d743597 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -401,7 +401,7 @@ (define-public calcmysky-qt5
 (define-public casacore
   (package
     (name "casacore")
-    (version "3.4.0")
+    (version "3.6.1")
     (source
      (origin
        (method git-fetch)
@@ -410,7 +410,7 @@ (define-public casacore
              (commit (string-append "v" version))))
        (sha256
         (base32
-         "05ar5gykgh4dm826xplj5ri5rw7znhxrvin2l67a3mjwfys7r2a0"))
+         "0ja0ss1cjfx9j2pnmqzr51ipxrfij7i2c4bq4nqkgaxfk5q447i5"))
        (file-name (git-file-name name version))))
     (build-system cmake-build-system)
     (arguments
@@ -458,7 +458,18 @@ (define-public casacore
             (lambda _
               (substitute* "build-tools/casacore_assay"
                 (("QSUBP=.*$") "QSUBP=\n")
-                (("YODP=.*$") "YODP=\n")))))))
+                (("YODP=.*$") "YODP=\n"))))
+          ;; XXX: It fails to find the stdlib types when the gfortran header
+          ;; is used.  Remove gfortran from CPLUS_INCLUDE_PATH as a
+          ;; workaround.  Taken from <https://issues.guix.gnu.org/73439#45>.
+          (add-after 'set-paths 'hide-gfortran
+            (lambda _
+              (let ((gfortran #$(this-package-input "gfortran")))
+                (setenv "CPLUS_INCLUDE_PATH"
+                        (string-join
+                         (delete (string-append gfortran "/include/c++")
+                                 (string-split (getenv "CPLUS_INCLUDE_PATH") #\:))
+                         ":"))))))))
     (native-inputs
      (list bison
            boost
@@ -469,6 +480,7 @@ (define-public casacore
            fftw
            fftwf
            gfortran
+           gsl
            hdf5
            ncurses
            openblas
-- 
2.47.1

